mlir/lib/Mitigation/Transforms/MitigationMethods/Zne.cpp

```diff
@@ -246,7 +246,11 @@ FlatSymbolRefAttr globalFolding(Location loc, PatternRewriter &rewriter, std::st
     Value c1 = rewriter.create<index::ConstantOp>(loc, 1);
     TypedAttr numberQubitsAttr = rewriter.getI64IntegerAttr(numberQubits);
     Value numberQubitsValue = rewriter.create<arith::ConstantOp>(loc, numberQubitsAttr);
-    rewriter.create<quantum::DeviceInitOp>(loc, lib, name, kwargs);
+
+    // TODO: clone all dependencies for the shots SSA value
```

Comment:
TODO: clone all dependencies for the shots SSA value

Reply:
Discussed in person, we will try to clone the entire chain with backslice

Reply:
(leaving this unresolved until I actually did it)

Reply:
Discussed, we realized since in the frontend calculation of shots will happen outside of the qnode, before `qml.device(..., shots = <some value computed earlier>)` is called, the SSA def-use computation chain will actually not be inside the qnode and hence using backward slice to clone the chain into the folded circuit is not what we want.

For now, we will simply create a `arith.constant` SSA shots value for the ZNE tests, just to not block ourselves. A potential design is take in the shots as a qnode argument, potentially tying into #1323 , recording the shots somehow in the qnode's own module abstraction. However that involves designing the entire frontend/mlir stack for dynamic shots and is a bit immature at this point.